### PR TITLE
Fix Caravel Issues

### PR DESCRIPTION
**Pull request by donn**
* `OpenROAD.CTS`
  * Made `CTS_MAX_CAP` a non-PDK value, and also optional as the values in the PDKs are bad and OpenROAD does a better job without it
  * CTS no longer passes `MAX_TRANSITION_CONSTRAINT`, instead using a new variable `CTS_MAX_SLEW` if it exists

* `Verilator.Lint`
  * Fixed issue where the order of files may not be preserved for macros

---

Issues reported by @mo-hosni

### CODE REVIEW

File: openlane/steps/openroad.py [hunk 1]
```diff
@@ -1806,10 +1806,15 @@ class CTS(ResizerStep):
             ),
             Variable(
                 "CTS_MAX_CAP",
-                Decimal,
-                "Defines the maximum capacitance, used in CTS.",
+                Optional[Decimal],
+                "Overrides the maximum capacitance used in CTS.",
```
Comment (kareefardi) [inline openlane/steps/openroad.py:1810]: If it overrides, where it the default value coming from?

File: openlane/steps/openroad.py [hunk 2]
```diff
@@ -1806,10 +1806,15 @@ class CTS(ResizerStep):
...
                 units="pF",
-                pdk=True,
```
Comment (kareefardi) [inline openlane/steps/openroad.py:1812]: why isn't it a PDK variable?

File: openlane/steps/openroad.py (hunk 1 above)
Comment (donn) [reply openlane/steps/openroad.py]: Added to description

File: openlane/steps/openroad.py (hunk 2 above)
Comment (donn) [reply openlane/steps/openroad.py:1812]: It's a user override- CTS can get the values from the liberty file, so no need to declare it as a PDK config

**Review by kareefardi — APPROVED** [review]

**Pull request merged**